src/providers/formattingProvider.ts

```diff
@@ -77,19 +94,71 @@ export class FormatProvider implements vscode.DocumentFormattingEditProvider {
 
             // This line
 
+            // Start directives for formatter
+            if (emptyLine) {
+                if (comment.match(/;\s*@AHK\+\+FormatBlockCommentOn/i)) {
+                    formatBlockComment = true;
+                }
+            }
+
             // Block comments
-            if (originalLine.match(/ *\/\*/)) {
+            // The /* and */ symbols can be used to comment out an entire section,
+            // but only if the symbols appear at the beginning of a line (excluding whitespace),
+            // as in this example:
+            // /*
+            // MsgBox, This line is commented out (disabled).
+            // MsgBox, Common mistake: */ this does not end the comment.
+            // MsgBox, This line is commented out.
+            // */
+            if (!blockComment && originalLine.match(/^\s*\/\*/)) {
+                // found start '/*' pattern
                 blockComment = true;
+                // Save first capture group (original indent)
+                blockCommentIndent = originalLine.match(/(^\s*)\/\*/)?.[1];
+                if (formatBlockComment) {
+                    // save indent values on block comment enter
+                    preBlockCommentDepth = depth;
+                    preBlockCommentTagDepth = tagDepth;
+                    preBlockCommentAtTopLevel = atTopLevel;
+                    preBlockCommentOneCommandCode = oneCommandCode;
+                    // reset indent values to default values with added current 'depth' indent
+                    oneCommandCode = false;
+                }
             }
-            if (originalLine.match(/ *\*\//)) {
-                blockComment = false;
-            }
+
             if (blockComment) {
-                formattedDocument += originalLine;
-                if (lineIndex !== document.lineCount - 1) {
-                    formattedDocument += '\n';
+                // Save block comment line only if user don't want format it content
```

Comment:
Can you expand on what this means? I don't follow "only if user don't want format it content" :/

Reply:
By default, the user does not want format text inside a block comment. The formatter stores the line directly in the result variable `formattedDocument` and immediately moves to the next line of the document.

If the user sets the directive `;@AHK++FormatBlockCommentOn`, he wants to format the text within the block comment. The variable `formatBlockComment` becomes `true` and the formatter does not save the line, but parses it like regular code.

Reply:
Please, suggest proper comment.